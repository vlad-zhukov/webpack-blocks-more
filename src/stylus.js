/*
 * Stylus loader.
 */
export default function stylus(options = {}) {
    const {extractOptions = {}, cssOptions = {}, stylusOptions = {}} = options;

    return (context, {addLoader}) =>
        addLoader({
            ...context.match,
            use: [
                {
                    loader: 'extract-loader',
                    options: extractOptions,
                },
                {
                    loader: 'css-loader',
                    options: cssOptions,
                },
                {
                    loader: 'stylus-loader',
                    options: {
                        compress: true,
                        'disable-cache': true,
                        preferPathResolver: 'webpack',
                        ...stylusOptions,
                    },
                },
            ],
        });
}
