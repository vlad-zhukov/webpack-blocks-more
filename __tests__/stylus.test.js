import {createConfig, match} from '@webpack-blocks/webpack';
import {file} from '@webpack-blocks/assets';
import stylus from '../src/stylus';

it('should work with match()', () => {
    const config = createConfig([match('*.styl', [stylus()])]);

    expect(config).toMatchSnapshot();
});

it('should take options', () => {
    const config = createConfig([
        match('*.styl', [
            stylus({
                stylusOptions: {
                    compress: false,
                },
            }),
        ]),
    ]);

    expect(config).toMatchSnapshot();
});

it('should work with file block', () => {
    const config = createConfig([match('*.styl', [file(), stylus()])]);

    expect(config).toMatchSnapshot();
});
