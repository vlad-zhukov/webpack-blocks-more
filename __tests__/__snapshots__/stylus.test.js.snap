// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take options 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.styl\\$/,
        "use": Array [
          Object {
            "loader": "extract-loader",
            "options": Object {},
          },
          Object {
            "loader": "css-loader",
            "options": Object {},
          },
          Object {
            "loader": "stylus-loader",
            "options": Object {
              "compress": false,
              "disable-cache": true,
              "preferPathResolver": "webpack",
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with file block 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.styl\\$/,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {},
          },
          Object {
            "loader": "extract-loader",
            "options": Object {},
          },
          Object {
            "loader": "css-loader",
            "options": Object {},
          },
          Object {
            "loader": "stylus-loader",
            "options": Object {
              "compress": true,
              "disable-cache": true,
              "preferPathResolver": "webpack",
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with match() 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.styl\\$/,
        "use": Array [
          Object {
            "loader": "extract-loader",
            "options": Object {},
          },
          Object {
            "loader": "css-loader",
            "options": Object {},
          },
          Object {
            "loader": "stylus-loader",
            "options": Object {
              "compress": true,
              "disable-cache": true,
              "preferPathResolver": "webpack",
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;
